Capri.Web/src/components/promoterView.vue

```diff
@@ -0,0 +1,262 @@
+                        thesis: 'Master',
+                        taken: 0,
+                        available: 3,
+                        type: 'full-time',
+                    },
+                    {
+                        title: 'Jan Szczuka',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Master',
+                        taken: 0,
+                        available: 3,
+                        type: 'full-time',
+                    },
+                    {
+                        title: 'Dakota Rice',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Master',
+                        taken: 0,
+                        available: 3,
+                        type: 'full-time',
+                    },
+                    {
+                        title: 'Dakota Rice',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Master',
+                        taken: 0,
+                        available: 3,
+                        type: 'full-time',
+                    },
+                    {
+                        title: 'Minerva Hooper',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Bachelor',
+                        taken: 3,
+                        available: 3,
+                        type: 'full-time',
+                    },
+                    {
+                        title: 'Sage Rodriguez',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Master',
+                        taken: 0,
+                        available: 3,
+                        type: 'full-time',
+                    },
+                    {
+                        title: 'Philip Chanley',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Bachelor',
+                        taken: 2,
+                        available: 3,
+                        type: 'part-time',
+                    },
+                    {
+                        title: 'Doris Greene',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Bachelor',
+                        taken: 0,
+                        available: 3,
+                        type: 'part-time',
+                    },
+                    {
+                        title: 'Mason Porter',
+                        // tslint:disable-next-line:object-literal-sort-keys
+                        thesis: 'Bachelor',
+                        taken: 2,
+                        available: 3,
+                        type: 'full-time',
+                    },
+                ],
+            };
+        }
+        public setColor(available, taken): string {
```

Comment:
Same thing as previosly. I'm thinking, why don't you create a component named like `ThesisTableComponent` with all these methods? You can use slots to pass data and customize it to view type.

https://vuejs.org/v2/guide/components-slots.html

Reply:
You won't be repeating yourself.

DRY principle says `Don't repeat yourself` ;)